plugins/poll/assets/javascripts/components/poll-breakdown-chart.js.es6

```diff
@@ -0,0 +1,175 @@
+import I18n from "I18n";
+import Component from "@ember/component";
+import { htmlSafe } from "@ember/template";
+import discourseComputed from "discourse-common/utils/decorators";
+import { PIE_CHART_TYPE } from "../controllers/poll-ui-builder";
+import { getColors } from "../lib/chart-colors";
+
+// args: options, group, displayMode, highlightedOption, setHighlightedOption
+export default Component.extend({
+  classNames: "poll-breakdown-chart-container",
+  optionToSlice: {},
+  previousHighlightedSliceIndex: null,
+  previousDisplayMode: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    const canvas = this.element.querySelector("canvas");
+    this.set(
```

Comment:
Since we're unlikely to observe or make a CP based on `chart` you might as well say `this.chart = ` instead of `set`. 

You could also consider giving `_chart` an underscore to indicate it's private unless you expect other people to use it in their code outside of this class.